abstract/group.go

```diff
@@ -51,6 +51,9 @@ type Secret interface {
 
 	// Set to a fresh random or pseudo-random secret
 	Pick(rand cipher.Stream) Secret
+
+	// SetBytes will take bytes and create a secret out of it
+	SetBytes([]byte) Secret
```

Comment:
Oh wait - by "a == a2" you meant a comparison of the *contents* of the slices, not a comparison of the slice-pointers?  (Because syntactically, "a == a2" in Go does *not* do a slice contents-comparison, if that's what you meant.)

If you did mean slice content-comparison, then yes, Marshal/Unmarshal can and should be expected to differ from SetBytes/Bytes in certain semantic subtleties.  

Marshal then Unmarshal then Marshal again should always produce the same thing the second time, since its purpose is to encode the internal state into its appropriate fixed-length "native" representation.  Similarly, if you do:

a := sec.Bytes()
sec.SetBytes(a)
a2 := sec.Bytes()

...then a byte wise comparison of a and a2 should always return true, even if the length of a (and a2) may vary depending on the specific integer contained in 'sec'.

However, UnmarshalBinary definitely should differ from SetBytes in that there are some inputs to SetBytes (e.g., 512-bit integers) for which a subsequent Bytes() won't yield the same thing.  Whereas if you do sec.UnmarshalBinary(a), then sec.MarshalBinary(), you should always get back a *if* a was actually a valid encoding of something.  (Note that sec.UnmarshalBinary() might return an error if not.)